x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/upgrade_prebuilt_rules_table/upgrade_prebuilt_rules_table_context.tsx

```diff
@@ -270,10 +271,21 @@ export const UpgradePrebuiltRulesTableContextProvider = ({
         {previewedRule && (
           <RuleDetailsFlyout
             rule={previewedRule}
-            actionButtonLabel={i18n.UPDATE_BUTTON_LABEL}
-            isActionButtonDisabled={canPreviewedRuleBeUpgraded}
-            onActionButtonClick={upgradeOneRule}
+            dataTestSubj="updatePrebuiltRulePreview"
             closeFlyout={closeRulePreview}
+            ruleActions={
+              <EuiButton
+                disabled={canPreviewedRuleBeUpgraded}
+                onClick={() => {
+                  upgradeOneRule(previewedRule.rule_id ?? '');
+                  closeRulePreview();
+                }}
+                fill
+                data-test-subj="updatePrebuiltRuleButton"
```

Comment:
Similar comment that I made for `add_prebuilt_rules_table_context.tsx`

Reply:
Makes sense, will change the `data-test-subj`. I think it should be fine without the `id` since there's only one install / upgrade button on the screen at a time.